cilium/cmd/bpf_egress_list.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cilium/cilium/pkg/command"
+	"github.com/cilium/cilium/pkg/common"
+	"github.com/cilium/cilium/pkg/maps/egressmap"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	egressListUsage = "List egress entries.\n"
+)
+
+var bpfEgressListCmd = &cobra.Command{
+	Use:     "list",
+	Aliases: []string{"ls"},
+	Short:   "List egress entries",
+	Long:    egressListUsage,
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("cilium bpf egress list")
+
+		bpfEgressList := make(map[string][]string)
+		if err := egressmap.EgressMap.Dump(bpfEgressList); err != nil {
+			fmt.Fprintf(os.Stderr, "error dumping contents of map: %s\n", err)
+			os.Exit(1)
+		}
+
+		if command.OutputJSON() {
+			if err := command.PrintOutput(bpfEgressList); err != nil {
+				fmt.Fprintf(os.Stderr, "error getting output of map in JSON: %s\n", err)
+				os.Exit(1)
+			}
+			return
+		}
+
+		if len(bpfEgressList) == 0 {
+			fmt.Fprintf(os.Stderr, "No entries found.\n%v\n", kernelVersionWarning)
```

Comment:
`kernelVersionWarning` has:
```
Note that for Linux kernel versions between 4.11 and 4.15 inclusive, the native
LPM map type used for implementing the IPCache does not provide the ability to
walk / dump the entries, so on these kernel versions this tool will never
return any entries, even if entries exist in the map. You may instead run:
    cilium map get cilium_ipcache
```
I think we need to adapt the message to refer to the egress map instead of the ipcache. Could be just a matter of passing it the map name:
```go
fmt.Sprintf(kernelVersionWarning, "egress map", egressmap.MapName)
```